### PR TITLE
Fixes to NWBExtractors

### PR DESCRIPTION
**Pull request by alejoe91**
@khl02007 Can you test if this fixes the timestamp issue?

@samuelgarcia also the nwbfile is only opened in the segment `__init__`

### CODE REVIEW

**Comment by samuelgarcia** [comment]
I think it should be open once and not in every segment.
I guess that io.read() do some checking about the file and hierachy.
So when we dump this and distribute it in 40 process io.read() is done 2 times. Once should be enought.
I really propose `self._nwbfile = io.read()` at recording level.

**Comment by alejoe91** [comment]
> I think it should be open once and not in every segment. I guess that io.read() do some checking about the file and hierachy. So when we dump this and distribute it in 40 process io.read() is done 2 times. Once should be enought. I really propose `self._nwbfile = io.read()` at recording level.

done

**Comment by khl02007** [comment]
@alejoe91 This seems to work - it returns timestamps upon `get_times`

**Comment by alejoe91** [comment]
@khl02007 note that I've taken the chance to improve the extractor as a whole. I think it should also make it faster for parallel processing :)

File: spikeinterface/extractors/nwbextractors.py [hunk 1]
```diff
@@ -54,118 +72,131 @@ class NwbRecordingExtractor(BaseRecording):
...
+        if hasattr(self._es, 'rate'):
+            sampling_frequency = self._es.rate
+        
+        if hasattr(self._es, 'starting_time'):
+            t_start = self._es.starting_time
+        else:
+            t_start = 0.
+        
+        timestamps = None
+        if hasattr(self._es, 'timestamps'):
```
Comment (samuelgarcia) [inline spikeinterface/extractors/nwbextractors.py:96]: and load_time_vector

Comment (alejoe91) [reply spikeinterface/extractors/nwbextractors.py:96]: no this is ok because only the object reference is accessed (and the correct `t_start`)

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**